lib/src/front_end/piece_writer.dart

```diff
@@ -98,33 +167,38 @@ class PieceWriter {
     return _currentText!;
   }
 
-  /// Ends the current text piece and (lazily) begins a new one.
+  /// Pops the previous piece and begins a new one.
```

Comment:
[nit] Between the two docs we describe the same piece as "the previous piece" and "the most recent piece". I think the latter makes the most sense.

```suggestion
  /// Pops the most recent piece and begins a new one.
```